gossip/tx_scrambler_test.go

```diff
@@ -0,0 +1,342 @@
+package gossip
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"golang.org/x/exp/rand"
+	"math"
+	"math/big"
+	"reflect"
+	"testing"
+)
+
+func TestTxScrambler_AnalyseEntryList_RemovesDuplicateTransactions(t *testing.T) {
+	entries := createScramblerTestInputRepeatedHashes(2, 10)
+	entries, _, _ = analyseEntryList(entries)
+
+	checkDuplicateHashes(t, entries)
```

Comment:
nit: This test would pass if the implementation of `analyseEntryList` would just always return an empty list.